CondFormats/HcalObjects/src/HcalConvertedPedestalWidthsGPU.cc

```diff
@@ -135,17 +135,12 @@ HcalConvertedPedestalWidthsGPU::HcalConvertedPedestalWidthsGPU(HcalPedestals con
   }
 }
 
-HcalConvertedPedestalWidthsGPU::Product::~Product() {
-  // deallocation
-  cudaCheck(cudaFree(values));
-}
-
 HcalConvertedPedestalWidthsGPU::Product const& HcalConvertedPedestalWidthsGPU::getProduct(
     cudaStream_t cudaStream) const {
   auto const& product = product_.dataForCurrentDeviceAsync(
       cudaStream, [this](HcalConvertedPedestalWidthsGPU::Product& product, cudaStream_t cudaStream) {
-        // malloc
-        cudaCheck(cudaMalloc((void**)&product.values, this->values_.size() * sizeof(float)));
+        // allocate
+        product.values = cms::cuda::make_device_unique<float[]>(values_.size(), cudaStream);
```

Comment:
> It follows that at the destruction time of Product the CUDA event is recorded on that CUDA stream, and the allocation is considered at live until that event becomes completed.

Do you think it would be better to use the default stream here ?

Reply:
> > Currently `cms::cuda::StreamCache` is a function-static, so it gets destroyed at the program end,

I was partially wrong, the `StreamCache` itself is destructed at the program end, but it is cleared by an explicit call from `CDUAService` destructor (to destroy the streams before resetting the CUDA runtime).

> > It follows that at the destruction time of Product the CUDA event is recorded on that CUDA stream, and the allocation is considered at live until that event becomes completed.
> 
> Do you think it would be better to use the default stream here ?

After some though I think the default stream would not help in the end. Already with `cudaMalloc()` and `cudaFree()` we essentially require that `Product` must be destructed before `CUDAService` is destructed (because of `cudaDeviceReset()` there). Therefore using `cms::cuda::device::unique_ptr` in `Product` would not really add any new constraints.

That leaves the "memory block considered being live for longer than necessary" the only potential issue, but doubt it would cause problems in practice (or we would risk having other problems because of running that close on device memory limit).

Reply:
OK, I'll leave the PR as it is.